mm2src/rpc_task/src/manager.rs

```diff
@@ -103,10 +102,26 @@ impl<Task: RpcTask> RpcTaskManager<Task> {
         Some(rpc_status)
     }
 
-    pub fn new_shared() -> RpcTaskManagerShared<Task> { Arc::new(Mutex::new(Self::default())) }
+    pub fn new(streaming_manager: StreamingManager) -> Self {
+        RpcTaskManager {
+            tasks: HashMap::new(),
+            streaming_manager,
+        }
+    }
```

Comment:
Shouldn't we make `StreamingManager` optional in `RpcTaskManager`?

Reply:
the optionality layer is provided by the client when they don't include their `client_id` in their request (will be changed tho to default to `0`). or could be provided by some individual task by returning `None` from `client_id()`.

on the manager (and even on task) level i think we really shouldn't make the streaming optional, as the normal interface for task managed rpcs are very dev-unfriendly.

put simply, we shouldn't have a task manager (responsible from some specific task pool) not support streaming all together, otherwise we could have made this task manager endpoint a set & forget rpc request instead.

Reply:
Ok, more simpler: If I want to turn off streaming completely, why am I having this instance at runtime?

Reply:
we can choose not to have it at runtime and save a couple of empty hashmaps of space but get some added type annotations and `if let`s everywhere we use the manager so to explicitly represent that the manager could be disabled (but exemplified over every usage of it)
OR
have it disabled right from the appropriate endpoint and keeping the struct representation intact.

I'm fine with either route tho :/

Reply:
> Why do we need to have if lets everywhere? Can you link some places where this field being utilized?

cuz the streaming manager is everywhere (or supposed to be accessible from anywhere). that's the point of it, the global thing that all the code has access to to send notifications & data to streamers running in the background. ctrc+f `streaming_manager|event_stream_manager` to see where it's being used.

Reply:
If it's not enabled, what happens then? Do we still send notification through this instance?